packages/dev/core/src/Shaders/ShadersInclude/pbrBlockReflectivity.fx

```diff
@@ -98,43 +98,59 @@ reflectivityOutParams reflectivityBlock(
+        #endif
 
-            // Compute the converted reflectivity.
-            surfaceReflectivityColor = mix(0.16 * reflectance * reflectance, baseColor, metallicRoughness.r);
+        #ifdef LEGACY_SPECULAR_ENERGY_CONSERVATION
+            outParams.surfaceAlbedo = baseColor.rgb * (vec3(1.0) - vec3(dielectricF0) * surfaceReflectivityColor) * (1.0 - outParams.metallic);
         #else
-            vec3 metallicF0 = metallicReflectanceFactors.rgb;
-
-            #if DEBUGMODE > 0
-                outParams.metallicF0 = metallicF0;
-            #endif
-
-            // Compute the converted diffuse.
-            outParams.surfaceAlbedo = mix(baseColor.rgb * (1.0 - metallicF0), vec3(0., 0., 0.), metallicRoughness.r);
-
-            // Compute the converted reflectivity.
-            surfaceReflectivityColor = mix(metallicF0, baseColor, metallicRoughness.r);
+            outParams.surfaceAlbedo = baseColor.rgb;
         #endif
+        
+        // Compute the coloured F0 reflectance.
+        // The coloured reflectance is the percentage of light reflected by the specular lobe at normal incidence.
+        // In glTF and OpenPBR, it is not the same thing as the percentage of light blocked from penetrating
+        // down to the diffuse lobe. The non-coloured F0 will be used for this (see below).
+        vec3 dielectricColorF0 = vec3(dielectricF0 * surfaceReflectivityColor);
+        vec3 metallicColorF0 = baseColor.rgb;
+        outParams.colorReflectanceF0 = mix(dielectricColorF0, metallicColorF0, outParams.metallic);
+
+        // Compute non-coloured reflectance.
+        // reflectanceF0 is the non-coloured reflectance used for blending between the diffuse and specular components.
+        // It represents the total percentage of light reflected by the specular lobe at normal incidence.
+        // In glTF's material model, the F0 value is multiplied by the maximum component of the specular colour.
+        dielectricF0 *= max(surfaceReflectivityColor.r, max(surfaceReflectivityColor.g, surfaceReflectivityColor.b));
+        outParams.reflectanceF0 = mix(dielectricF0, 1.0, outParams.metallic);
+
+        #ifdef LEGACY_SPECULAR_ENERGY_CONSERVATION
+            surfaceReflectivityColor = mix(surfaceReflectivityColor, baseColor.rgb, outParams.metallic);
+            outParams.reflectanceF0 = max(surfaceReflectivityColor.r, max(surfaceReflectivityColor.g, surfaceReflectivityColor.b));
+        #endif
+        
+        // Scale the reflectanceF90 by the IOR for values less than 1.5.
+        // This is an empirical hack to account for the fact that Schlick is tuned for IOR = 1.5
+        // and an IOR of 1.0 should result in no visible glancing specular.
+        float f90Scale = clamp(2.0 * (ior - 1.0), 0.0, 1.0);
+        #ifdef LEGACY_SPECULAR_ENERGY_CONSERVATION
+            outParams.reflectanceF90 = vec3(mix(specularWeight * f90Scale, specularWeight, outParams.metallic));

```

Comment:
I think that this one should be:
```glsl
outParams.reflectanceF90 = vec3(specularWeight);
```
to be fully backward compatible?

Reply:
Done